### ISSUE TITLE
Incorrect typescript errors produced when `declaration` and `noEmit` are both `true`

### ISSUE THREAD

**Issue by bendemboski** [microsoft/TypeScript#52927]
- VS Code Version: 1.75.1
- OS Version: MacOS 13.1

Steps to Reproduce:

1. Create a typescript project with a single `.ts` file containing the following:
```ts
class Foo {
  private foo = 'FOO';
}

function bar(value: string) {
  class Bar extends Foo {
    bar = value;
  }

  return new Bar();
}
```
and a `tsconfig.json` with:
```json
{
  "compilerOptions": {
    "declaration": true,
    "noEmit": true,
  },
}
```
2. Run `tsc` from the command-line and observe no errors
3. Open project in VSCode and open the `.ts` file, and observe that it flags a `ts(4094)` error

It looks like when running `tsc`, the `noEmit` option "overrides" the `declaration` option and prevents the generation of declaration files, which means declaration-related errors like `ts(4094)` do not occur. VSCode appears to be ignoring the `noEmit` option in this case and incorrectly producing declaration-related errors.

**Comment by akhil888binoy** [comment]
The issue you are experiencing is likely caused by VS Code using its own TypeScript version that is different from the one installed on your system. This can sometimes lead to inconsistencies between the VS Code TypeScript language service and the TypeScript compiler used by tsc.

To resolve the issue, you can try updating the TypeScript version used by VS Code by following these steps:

Open VS Code and navigate to the Extensions panel on the left-hand side.
Search for "TypeScript" and click on the "TypeScript and JavaScript Language Features" extension.
Click on the gear icon to open the extension settings.
Under "Workspace Settings", check the "TypeScript Path" setting and make sure it points to the correct path of your installed TypeScript version.
If the TypeScript version used by VS Code is outdated, you can update it by running "npm install -g typescript" in your terminal.
After updating the TypeScript version used by VS Code, try reopening your TypeScript project and see if the error is still present. If the issue persists, you can try disabling the built-in TypeScript extension in VS Code and instead using the TypeScript version installed on your system. To do this, follow these steps:

Open VS Code and navigate to the Extensions panel on the left-hand side.
Search for "TypeScript" and click on the "TypeScript" extension.
Click on the gear icon to open the extension settings.
Under "Extension Settings", uncheck the "Use Workspace Version" setting.
Restart VS Code and reopen your TypeScript project.
By disabling the built-in TypeScript extension and using the TypeScript version installed on your system, you can ensure that VS Code and tsc are using the same TypeScript version, which should help resolve any inconsistencies or errors.

**Comment by philly-vanilly** [comment]
I had the same type of (confusing) error with typescript 5, for me it was on an Intellij IDE running via terminal (so completely VSCode unrelated) trying to start a Vite dev-server:

```
 ERROR(TypeScript)  Composite projects may not disable declaration emit.
 FILE  undefined:

undefined

[TypeScript] Found 1 error. Watching for file changes.
```
I have spent a lot of time looking at various tscconfig properties distributed over many related tsconfigs in my monorepo (as it uses composite: true) before I found the declaration property to be responsible. A more complete error hint would have been helpful

**Comment by RyanCavanaugh** [comment]
Closing language service bugs related to the 6.0 implementation. For more information, see #62827
